### PR TITLE
feat: Enable_VPC_native_for_clusters

### PR DESCRIPTION
**Pull request by AnjaliMadhavi-Nakirikanti**
# Description

Enable_VPC_native_for_clusters
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The commit message/PR follows our guidelines

# **Other information**:

### CODE REVIEW

File: jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/file/AssetFileGenerator.java [hunk 1]
```diff
@@ -34,41 +34,41 @@ public class AssetFileGenerator {
...
-	PubSubInventoryCollector pubSubInventoryCollector;
-
-	@Autowired
-	KmsKeyInventoryCollector kmsKeyInventoryCollector;
-
-	@Autowired
-	DataProcInventoryCollector dataProcInventoryCollector;
+//	@Autowired
+//	VMInventoryCollector vmInventoryCollector;
+//	@Autowired
```
Comment (santhosh-challa) [inline jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/file/AssetFileGenerator.java:39]: I think this commented code was left out post-testing. Can you please check?

**Review by santhosh-challa — CHANGES_REQUESTED** [review]

File: jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/file/AssetFileGenerator.java [hunk 2]
```diff
@@ -196,46 +196,46 @@ public void generateFiles(List<ProjectVH> projects, String filePath) {
...
+//					return;
+//				}
+//				try {
+//					FileManager.generateServiceAccountFiles(serviceAccountInventoryCollector.fetchServiceAccountDetails(project));
+//				} catch (Exception e) {
+//					e.printStackTrace();
+//				}
+//			});
 
 			executor.shutdown();
```
Comment (sudhirrs) [inline jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/file/AssetFileGenerator.java:241]:
```while (!executor.isTerminated()) {
			}```
			these kind of checks can hog cpu. Please add a sleep inside even if for 500ms.

File: jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/file/AssetFileGenerator.java (hunk 1 above)
Comment (AnjaliMadhavi-Nakirikanti) [reply jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/file/AssetFileGenerator.java:39]: sure @santhosh-challa

File: installer/resources/pacbot_app/files/DB.sql [hunk 3]
```diff
@@ -1720,6 +1720,7 @@ INSERT IGNORE INTO cf_Policy (`policyId`,`policyName`,`policyDesc`,`resolution`,
...
+INSERT IGNORE INTO cf_Policy (`policyId`,`policyName`,`policyDesc`,`resolution`,`policyUrl`,`policyVersion`,`status`,`userId`,`createdDate`,`modifiedDate`,`resolutionUrl`) VALUES ('Enable_VPC_native_for_clusters','Enable VPC-native for clusters','Create Alias IPs for the node network CIDR range in order to subsequently configure IP-based policies and firewalling for pods. A cluster that uses Alias IPs is called a 'VPC-native' cluster.',NULL,NULL,'version-1',NULL,NULL,'2022-10-27','2022-10-27','https://github.com/PaladinCloud/CE/wiki/GCP-Policy#Enable-VPC-native-for-clusters');
```
Comment (dheerajkholia-paladin) [inline installer/resources/pacbot_app/files/DB.sql:1723]: Query has syntax error, please escape quotes.

File: jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/collector/GKEClusterInventoryCollector.java [hunk 4]
```diff
@@ -69,12 +69,7 @@ public List<GKEClusterVH> fetchGKEClusterInventory(ProjectVH project) throws IOE
                     gkeClusterVH.setProjectName(project.getProjectName());
                     gkeClusterVH.setProjectId(project.getProjectId());
                     gkeClusterVH.set_cloudType(InventoryConstants.CLOUD_TYPE_GCP);
-                    gkeClusterVH.setId(cluster.getId());
-                    gkeClusterVH.setProjectName(project.getProjectName());
-                    gkeClusterVH.setProjectId(project.getProjectId());
-                    gkeClusterVH.set_cloudType(InventoryConstants.CLOUD_TYPE_GCP);
-
-                    gkeClusterVH.setRegion(cluster.getLocation());
```
Comment (dheerajkholia-paladin) [inline jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/collector/GKEClusterInventoryCollector.java:77]: Please confirm if this deletion is not accidental.

File: jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/file/AssetFileGenerator.java (hunk 2 above)
Comment (santhosh-challa) [reply jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/file/AssetFileGenerator.java]: We will add this. thanks for the comment.

File: jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/collector/GKEClusterInventoryCollector.java (hunk 4 above)
Comment (AnjaliMadhavi-Nakirikanti) [reply jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/collector/GKEClusterInventoryCollector.java:77]: there are not actually removed these line got added above line 69

**Review by dheerajkholia-paladin — APPROVED** [review]

**Review by santhosh-challa — APPROVED** [review]

**Pull request merged**